mypy/newsemanal/semanal.py

```diff
@@ -2634,7 +2634,9 @@ def process_typevar_parameters(self, args: List[Expression],
                     analyzed = self.expr_to_analyzed_type(param_value,
                                                           report_invalid_types=False)
                     if analyzed is None:
-                        return None
+                        # It is fine to simply use a temporary Any because we don't need the bound
+                        # for anything before main pass of semantic analysis is finished.
+                        analyzed = AnyType(TypeOfAny.special_form)
```

Comment:
Update comment to mention that we will incrementally populate `TypeVarExpr` if some part is missing.

However, do we always defer somewhere if we get here? In particular, there should probably be a test case where this is the only cause of deferral, if possible. 

Previously the caller deferred if there is a `None` return from here -- which actually doesn't look quite right in case of some the above `None` returns. What happens if one of the above checks fails (e.g., `covariant` with invalid value)?

Reply:
Thanks! I removed the unnecessary `mark_incomplete()` call from the caller, and added a detailed comment explaining why it is safe to not defer here.